### ISSUE TITLE
static preprocessing gone wrong?

### ISSUE THREAD

**Issue by robinkraft** [wri/forma-clj#202]
### TL;DR

Of the 60m VCF >= 25 pixels we're interested in, 5.6 million actually have pixel values that suggest they're water, not forest (i.e. all -3000 fill values). Upon actually mapping some of them, I found that they are actually in water. This doesn't seem to have happened in the past, otherwise all of our data would be really off. It's not a reprojection thing, but there's some major shifting happening early on in the static data processing. Read on for more.
### Reprojection is ok

Here's the `"500" 13 11 0 0` pixel in SE Brazil. The dot a bit down and to the left of the other one is the calculated centroid (using the `reproject` namespace). The latlon is `-20.00208333333334 -53.207375785975294`. The other is the centroid I picked by hand with the mouse.

![](https://img.skitch.com/20120820-x9jhemg6k9727jfdh5mqm2ku2s.jpg)

Looks good to me. I also checked the bottom left and top right - all good. The shifting I'm seeing with VCF values is hundreds of kilometers.
### The data

Below you can see the calculated tile centroid in the upper left corner of a raw NDVI raster layer. And in the bottom right there's a latlon coordinate for reference. The yellowish raster is the raw VCF file. The outline is GADM for Brazil, shifted slightly from the raw VCF due to different coordinate systems. The red/blue one is a raster derived from the latlons of all the VCF >= 25 pixels we have, as of just before the trends analysis.
### The big shift

The green dots halfway down - one near the coast, the other out in the middle of the water - should be in the same place on the map, but here are ~526 km apart. In any case, you can see that the shape of the coastline is shifted way east. The pixels in the water are also missing ecoregion and GADM ids - as you'd expect for watery pixels.

![](https://img.skitch.com/20120821-mdc3iti6s1a6d8wxsgbs9qbar.jpg)

The second image is an incomplete rendering of the VCF >= 25 pixels that come directly out of the pail, using calculated latlons to put them on the map. Arc was still drawing points, so you can only see some of them, but you can see that they line up with the earlier VCF pixels from our data.
### Conclusions?

Since the problem appears early on, when the static datasets are processed, we know that the problem must be happening there. I don't know yet whether this has been an issue in the past, but given that there are so few recent commits to the relevant namespaces ([static](https://github.com/reddmetrics/forma-clj/commits/develop/src/clj/forma/source/static.clj) and [preprocess](https://github.com/reddmetrics/forma-clj/commits/develop/src/clj/forma/hadoop/jobs/preprocess.clj)), that seems unlikely. I'll check it out though.

![](https://img.skitch.com/20120821-8g52rqaes7q3sepuhcbnj17k6m.jpg)
